data/cache/impl.py

```diff
@@ -93,6 +111,16 @@ def retrieve(self, cache_key, loader, should_cache=is_not_none):
 _JSON_TYPE = 2
 
 
+@contextmanager
+def memcache_client(client_builder):
+    try:
+        client = client_builder()
+        yield client
+    finally:
```

Comment:
Doesn't this have to `yield` at every codepath? If an exception is raised before `yield client`, then you'll get: `TypeError: 'NoneType' object is not an iterator`.

Reply:
https://amir.rachum.com/blog/2017/03/03/generator-cleanup/

Reply:
I'll add one in the finally

Reply:
```python 
@contextmanager
def memcached_client(client_builder):
    client = None
    try:
        client = client_builder()
    finally:
        yield client
        if client is not None:
            client.cloes()
```